### PR TITLE
[Transaction] Fix direct memory leak related to commit and abort markers

### PR DESCRIPTION
**Pull request by lhotari**
### Motivation

- Netty ByteBufs for commit and abort markers currently leak direct memory

- `ManagedLedgerImpl.asyncAddEntry` will increase the reference count by calling `.retain()` for the marker message's ByteBuf (in `OpAddEntry.create`) and therefore the reference count has to be decreased with the call to `.release()` after  the `asyncAddEntry` call.

### Modifications

- call `.release()` in try-finally block for commit and abort markers after the `asyncAddEntry` call

### CODE REVIEW

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/transaction/buffer/impl/TopicTransactionBuffer.java [hunk 1]
```diff
@@ -206,24 +206,28 @@ private void handleTransactionMessage(TxnID txnId, Position position) {
...
-        topic.getManagedLedger().asyncAddEntry(commitMarker, new AsyncCallbacks.AddEntryCallback() {
-            @Override
-            public void addComplete(Position position, ByteBuf entryData, Object ctx) {
-                synchronized (TopicTransactionBuffer.this) {
-                    updateMaxReadPosition(txnID);
-                    handleLowWaterMark(txnID, lowWaterMark);
-                    clearAbortedTransactions();
-                    takeSnapshotByChangeTimes();
+        try {
+            topic.getManagedLedger().asyncAddEntry(commitMarker, new AsyncCallbacks.AddEntryCallback() {
```
Comment (eolivelli) [inline pulsar-broker/src/main/java/org/apache/pulsar/broker/transaction/buffer/impl/TopicTransactionBuffer.java:210]: should we release commitMarker in this thread or inside the callback ?

Comment (lhotari) [reply pulsar-broker/src/main/java/org/apache/pulsar/broker/transaction/buffer/impl/TopicTransactionBuffer.java:210]: in this thread.

**Review by eolivelli — APPROVED** [review]
LGTM

**Review by codelipenghui — APPROVED** [review]
Nice catch.

**Review by sijie — APPROVED** [review]

**Comment by codelipenghui** [comment]
@lhotari I have removed the `release/2.7.4` tag for this PR since for branch-2.7, the transaction is just a developer preview version, some features does not complete.

**Pull request merged**